### ISSUE TITLE
How to make and test a Steam build?

### ISSUE THREAD

**Issue by SwamiUmami** [opensoldat/opensoldat#86]
When fixing #36 I wanted to make a Steam build of server, but I was not successful. First of all, I encountered compilation problems, that I temporarily fixed in [my branch](https://github.com/SwamiUmami/soldat/tree/fix-steam-compilation). After fixing this, I had problems when linking client and server binaries. More specifically, it couldn't find `-lsteam_api`. After scrolling through [documentation](https://partner.steamgames.com/doc/sdk/api) I downloaded the SteamWorks SDK, but it only offers version 1.53a, which seems incompatible with Soldat. Do I need a Steam account to download other versions? Can anyone make Steam builds, or is it restricted to members of THD, or how does it work? And more importantly, how can we test Steam builds? Does it also require a Steam account?
All of this raises many questions about Steam builds, but I can't find much documentation.

**Comment by BranDougherty** [comment]
Yeah I could never get it to work myself, maybe @helloer can fill us in if we're missing a few pieces of the puzzle.
What I have so far is:

1. Sign up for a Steam account
2. Log in to Steamworks
3. Sign Steamworks SDK Access agreements
4. Download appropriate SDK (you can just change the version at the end of the download URL, I believe we want https://partner.steamgames.com/downloads/steamworks_sdk_150.zip) (Edit: Oh actually they're listed here: https://partner.steamgames.com/downloads/list)
5. Install and run Steam on the machine you want to test on (must be actively running)
6. Build Soldat with -DSTEAM
7. Copy `sdk/redistributable_bin/linux64/libsteam_api.so` to the directory of the soldat executables.
8. Run soldat like normal

I also had to make the same changes to compile, and in addition I had to insert calls to `SteamAPI_Init()` and `SteamAPI_Shutdown` at the appropriate places in `Server.pas` to get the server to run. I'm not able to run the client, for some reason `SteamAPI_SteamNetworkingSockets_v009` is returning null while all the other interfaces are working (only in the client).
